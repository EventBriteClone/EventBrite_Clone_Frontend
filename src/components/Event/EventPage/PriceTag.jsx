import styles from "./PriceTag.module.css";
import LikeButton from "../../UI/LikeButton";
export default function PriceTag(props) {
  return (
    <div>
      {/* <LikeButton /> */}
      <div className={styles["price-tag"]}>
        <div className={styles["price"]}>{props.price}</div>
        <button
          id="eventbrite-widget-modal-trigger-588359929387"
          data-testid="checkout-link"
          data-tracking-label="Tickets"
          data-heap-id="Conversion Bar - Checkout Button"
          className={`${styles["eds-btn"]} ${styles["eds-btn--button"]} ${styles["eds-btn--fill"]} ${styles["js-embed-ticket-modal-btn"]}`}
          type="button"
        >
          Get tickets
        </button>
      </div>
    </div>
  );
}
