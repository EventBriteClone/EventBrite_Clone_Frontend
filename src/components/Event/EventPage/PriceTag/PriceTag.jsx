import React, { useState } from 'react';
import styles from "./PriceTag.module.css";
import Popup from "../../popup/Popup";
import { Link } from "react-router-dom"

export default function PriceTag(props) {
  const [show,setShow]=useState(false);
  const price = props.price;
  const image = props.img;
  return (
    <div>
      <div className={styles["price-tag"]}>
        <div className={styles["price"]}>{props.price}</div>
        <button
          id="eventbrite-widget-modal-trigger-588359929387"
          data-testid="checkout-link"
          data-tracking-label="Tickets"
          data-heap-id="Conversion Bar - Checkout Button"
          className={ `${styles["eds-btn"]} ${styles["eds-btn--button"]} ${styles["eds-btn--fill"]} ${styles["js-embed-ticket-modal-btn"]}`        }
          onClick={() => setShow(true)}
        >
          Get Tickets
        </button>
      </div>
      {show && <Popup price={price} event={props.id} show={show} setShow={setShow} img={image} title={props.title}/>}
    </div>
  );
}
