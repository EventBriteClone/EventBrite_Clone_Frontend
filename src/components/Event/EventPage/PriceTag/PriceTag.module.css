.price-tag {
  position: relative;
  width: 360px;
  height: 150px;
  margin-left: 1000px;
  margin-top: 70px;
  border-width: 1px;
  border-style: ridge;
  border-radius: 15px 15px 15px 15px;
}

@media screen and (max-width: 86em) {
  .price-tag {
    width: 80% !important;
  }
}

.price {
  margin-top: 20px;
  margin-left: 120px;
  width: 312px;
  height: 22.5px;
  font-size: 20px;
}

.eds-btn {
  font-size: 15px;
  color: white;
  margin-top: 30px;
  margin-left: 25px;
  -webkit-transform: translateZ(0);
  transform: translateZ(0);
  position: relative;
  width: 312px;
  height: 44px;
  padding: 0 30px 1px;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  text-align: center;
  text-decoration: none;
  line-height: 24px;
  font-weight: 600;
  letter-spacing: 0.2px;
  background: transparent;
  border: 2px solid #a9a8b3;
  border: 2px solid var(--eds-ui-500, #a9a8b3);
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.4s cubic-bezier(0.4, 0, 0.3, 1);
  transition: all 0.4s cubic-bezier(0.4, 0, 0.3, 1);
}

.eds-btn--button {
  background-color: orange;
  border-color: #d1410c;
  border-color: var(--eds-primary-brand, #d1410c);
}

.eds-btn--fill {
  background-color: #d1410c;
  border-color: #d1410c;
  border-color: var(--eds-primary-brand, #d1410c);
}

.eds-btn--fill:hover {
  opacity: 0.8;
}

.eds-btn--unavailable {
  background-color: #ccc;
  border-color: #ccc;
  cursor: default;
}

.eds-btn--unavailable {
  background-color: #ccc;
  border-color: #ccc;
  cursor: default;
}

.eds-btn--unavailable:hover {
  opacity: 1;
  cursor: default;
}