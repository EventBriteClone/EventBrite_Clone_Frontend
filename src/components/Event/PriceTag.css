.price-tag {
    position: sticky;
    width: 360px;
    height: 150px;
    margin-left: 1000px;
    margin-top: 70px;;
    border-width: 1px;
    border-style: ridge;
    border-radius: 15px 15px 15px 15px;
}

.price {
    margin-top:20px;
    margin-left:120px;
    width: 312px;
    height: 22.5px;
    font-size: 20px;
    font-family:'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.eds-btn {
    font-size: 15px;
    background-color: orange;
    color:white;
    margin-top: 30px;
    margin-left: 25px;;
    -webkit-transform: translateZ(0);
    transform: translateZ(0);
    position: relative;
    width:312px;
    height: 44px;
    padding: 0 30px 1px;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    text-align: center;
    text-decoration: none;
    line-height: 24px;
    font-weight: 600;
    letter-spacing: .2px;
    background: transparent;
    border: 2px solid #a9a8b3;
    border: 2px solid var(--eds-ui-500,#a9a8b3);
    border-radius: 4px;
    cursor: pointer;
    -webkit-transition: all .4s cubic-bezier(.4,0,.3,1);
    transition: all .4s cubic-bezier(.4,0,.3,1);
}

.eds-btn--fill {
    border-color: #d1410c;
    border-color: var(--eds-primary-brand,#d1410c);
    background-color: #d1410c;
    background-color: var(--eds-primary-brand,#d1410c);
}