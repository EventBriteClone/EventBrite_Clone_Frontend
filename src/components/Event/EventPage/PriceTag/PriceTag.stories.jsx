import PriceTag from "./PriceTag";
import styles from "./PriceTag.module.css";

export default {
  title: "Components/Event Page/Price Tag",
  component: PriceTag,
};

export const Primary = (args) => <PriceTag {...args} />;

Primary.args = { price: "$$$$$$$$$$",
buttonClass: `${styles["eds-btn"]} ${styles["eds-btn--button"]} ${styles["eds-btn--fill"]} ${styles["js-embed-ticket-modal-btn"]}`
 };

export const Unavailable = (args) => (
    <PriceTag
      {...args}
      buttonClass={`${styles["eds-btn"]} ${styles["eds-btn--button"]} ${styles["eds-btn--fill"]} ${styles["js-embed-ticket-modal-btn"]} ${styles["eds-btn--unavailable"]}`}
    />
  );
  