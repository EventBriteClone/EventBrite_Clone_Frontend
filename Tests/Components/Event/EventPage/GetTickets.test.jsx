import PriceTag from "../../../../src/components/Event/EventPage/PriceTag/PriceTag";
import { test, expect, describe } from "vitest";
import { create } from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("PriceTag", () => {
  it("check get tickets button", () => {
    const component = create(
      <MemoryRouter>
        <PriceTag />
      </MemoryRouter>
    );
    const buttons = component.root.findAll((el) => el.type === 'button');

    expect(buttons.length).toBe(1);
  });
});
