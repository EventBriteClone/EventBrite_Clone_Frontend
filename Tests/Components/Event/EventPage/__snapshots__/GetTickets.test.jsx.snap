// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`follow button > renders text 1`] = `
<div>
  <div
    className="_price-tag_601000"
  >
    <div
      className="_price_601000"
    >
      €549 – €699
    </div>
    <button
      className="_eds-btn_601000 _eds-btn--button_601000 _eds-btn--fill_601000 _js-embed-ticket-modal-btn_601000"
      data-heap-id="Conversion Bar - Checkout Button"
      data-testid="checkout-link"
      data-tracking-label="Tickets"
      id="eventbrite-widget-modal-trigger-588359929387"
      type="button"
    >
      Get tickets
    </button>
  </div>
</div>
`;
